### PR TITLE
Add semigroup instance for Void  #1777

### LINKED ISSUE #1777: Add Semigroup instance for Void

### PR DESCRIPTION
**Pull request by wi101**
Fixes #1777

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/data/Void.scala [hunk 1]
```diff
@@ -31,8 +32,17 @@ object VoidModule extends VoidSyntax {
   implicit def void_<~<[A]: Void <~< A = Void.conforms[A]
 }
 
+trait VoidInstance {
```
Comment (jdegoes) [inline base/shared/src/main/scala/scalaz/data/Void.scala]: For consistency with other examples, this should be called `VoidInstances`.

**Comment by wi101** [comment]
There is a compile error because the implicit val voidEq is moved in `Void`, there is a val Iso in `Prelude.scala` which uses this implicit Eq[Void] (package scalaz extends BaseTypeclasses ... and `Iso` is in BaseTypeclasses trait). I change Iso to a lazy val, the test works again.
@jdegoes What do you think? Do I keep this change or move Eq[Void] to EqInstances?

**Comment by jdegoes** [comment]
@wi101 I am good with either way (either keep the lazy val, or do not move `Eq[Void]` into `VoidInstances`). I think the question of how to solve these types of problems in general is undecided (#1763).

**Comment by wi101** [comment]
Ok, I'll do the second ;)

**Comment by jdegoes** [comment]
Thanks for the pull request! 🙏 😄

**Pull request merged**